components/camel-rest-swagger/src/main/java/org/apache/camel/component/rest/swagger/RestSwaggerEndpoint.java

```diff
@@ -592,31 +619,53 @@ static String hostFrom(final RestConfiguration restConfiguration) {
      * @return the specification
      * @throws IOException
      */
-    static Swagger loadSpecificationFrom(final CamelContext camelContext, final URI uri) throws IOException {
+    static Swagger loadSpecificationFrom(final CamelContext camelContext, final URI uri, SSLContextParameters sslContextParameters) throws IOException {
         final ObjectMapper mapper = Json.mapper();
 
         final SwaggerParser swaggerParser = new SwaggerParser();
 
         final String uriAsString = uri.toString();
 
-        try (InputStream stream = ResourceHelper.resolveMandatoryResourceAsInputStream(camelContext, uriAsString)) {
-            final JsonNode node = mapper.readTree(stream);
+        if (sslContextParameters == null) {
+            try (InputStream stream = ResourceHelper.resolveMandatoryResourceAsInputStream(camelContext, uriAsString)) {
+                return parseInputStream(swaggerParser, mapper, stream);
+            } catch (final Exception e) {
+                return loadSpecificationFallback(swaggerParser, uriAsString, e);
+            }
+        }
 
-            return swaggerParser.read(node);
+        HttpComponent httpComponent = new HttpComponent();
+        httpComponent.setSslContextParameters(sslContextParameters);
+        httpComponent.setCamelContext(camelContext);
+        try (HttpEndpoint e = (HttpEndpoint)httpComponent.createEndpoint(uriAsString);
+             HttpProducer p = (HttpProducer)e.createProducer();
+             InputStream stream = p.getHttpClient().execute(new HttpGet(uri)).getEntity().getContent())  {
```

Comment:
You likely need to close this input stream after use.

Reply:
@davsclaus I thought that try-with-resource will take care of closing. (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html). Am I missing something?

Reply:
It only takes care the stuff inside the try ( xxxx ) line, which in your case is creating an http endpoint

Reply:
oh, ma bad, I'll fix it

Reply:
@davsclaus Just to be sure, whole 
HttpEndpoint e = (HttpEndpoint)httpComponent.createEndpoint(uriAsString);
             HttpProducer p = (HttpProducer)e.createProducer();
             InputStream stream = p.getHttpClient().execute(new HttpGet(uri)).getEntity().getContent()

is inside try ( xxx ), so it should be all closed automatically. All 3 instances (e, p, stream) should be closed.

Reply:
Ah gotcha yeah its multi lined and in one giant ( ). Okay usually its 99% just one resource. Yeah it will be auto-closed.